### PR TITLE
Add alert and near full counters to gluster dashboard datapoints

### PR DESCRIPTION
**Pull request by anmolbabu**
Signed-off-by: anmolbabu <anmolbudugutta@gmail.com>

### CODE REVIEW

**Comment by anmolbabu** [comment]
Not yet verified....

**Comment by anmolbabu** [comment]
@nthomas-redhat @shtripat @r0h4n Please review

File: tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py [hunk 1]
```diff
@@ -90,15 +127,57 @@ def get_cluster_summary(self, cluster_id, cluster_det):
...
+    def get_system_brick_status_wise_counts(self, cluster_summaries):
+        brick_status_wise_counts = {}
+        brick_critical_alerts = []
+        brick_warning_alerts = []
+        for cluster_summary in cluster_summaries:
+            if self.name in cluster_summary.sds_type:
+                cluster_brick_count = cluster_summary.sds_det.get(
+                    'brick_status_wise_counts', {}
+                )
+                if isinstance(cluster_brick_count, unicode):
```
Comment (shtripat) [inline tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py]: In what scenario cluster_brick_count is unicode?? Add some comments may be.

File: tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py [hunk 2]
```diff
@@ -109,8 +188,24 @@ def get_system_volume_status_wise_counts(self, cluster_summaries):
                         cluster_volume_count.encode('ascii', 'replace')
                     )
                 for status, count in cluster_volume_count.iteritems():
-                    volume_status_wise_counts[status] = \
-                        volume_status_wise_counts.get(status, 0) + int(count)
+                    if isinstance(count, int):
```
Comment (shtripat) [inline tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py]: Why we need this isinstance check explicitly? Shouldn't `count` always be an `int`

File: tendrl/performance_monitoring/utils/__init__.py [hunk 3]
```diff
@@ -71,3 +73,36 @@ def read(key):
...
+        )
+    if resource_classification == pm_consts.CLUSTER:
+        alerts = NS.central_store_thread.get_cluster_alerts(
+            **kwargs
+        )
+    critical_alerts = []
+    warning_alerts = []
+    for alert in alerts:
+        if (
+            alert['acked'] == "True" or
```
Comment (shtripat) [inline tendrl/performance_monitoring/utils/__init__.py]: You can check as below

```
alert['acked'].lower() == 'true'
```

File: tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py (hunk 1 above)
Comment (anmolbabu) [inline tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py]: this is something to do with obj#save so I have this conditional...

File: tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py (hunk 2 above)
Comment (anmolbabu) [inline tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py]: There are other tags like list of relevant alerts(the only non-int for now...)

File: tendrl/performance_monitoring/sds/__init__.py [hunk 4]
```diff
@@ -61,7 +62,14 @@ def compute_system_summary(self, cluster_summaries, clusters):
...
     def get_clusters_status_wise_counts(self, clusters):
-        clusters_status_wise_counts = {'total': 0}
+        clusters_status_wise_counts = {
+            'total': 0,
+            'near_full': 0,
+            pm_consts.CRITICAL_ALERTS: [],
+            pm_consts.WARNING_ALERTS: []
+        }
+        cluster_critical_alerts = []
+        cluster_critical_alerts = []
```
Comment (anmolbabu) [inline tendrl/performance_monitoring/sds/__init__.py]: Done

File: tendrl/performance_monitoring/utils/__init__.py (hunk 3 above)
Comment (anmolbabu) [inline tendrl/performance_monitoring/utils/__init__.py]: Done

File: tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py (hunk 1 above)
Comment (shtripat) [inline tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py]: ok. Add some comments to tell the same.

File: tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py (hunk 2 above)
Comment (shtripat) [inline tendrl/performance_monitoring/sds/glusterfs/glusterfs_plugin.py]: ok. But somehow I feel it confusing. May be can think of something later and clean

**Comment by shtripat** [comment]
LGTM

**Pull request merged**